app/services/proxy_status/puppetca.rb

```diff
@@ -0,0 +1,71 @@
+module ProxyStatus
+  class PuppetCA < Base
+    def certs
+      fetch_proxy_data('/certs') do
+        api.all.map do |name, properties|
+          SmartProxies::PuppetCACertificate.new([name.strip, properties['state'], properties['fingerprint'], properties["not_before"], properties["not_after"], self])
+        end.compact.sort_by!(&:valid_from)
+      end
+    end
+
+    def find(name)
+      certs.find{|c| c.name == name}
+    end
+
+    def find_by_state(state)
+      certs.select{|c| c.state == state}
+    end
+
+    def expiry
+      ca_cert = certs.first
+      if ca_cert.present?
+        ca_cert.expires_at
+      else
+        _("Could not locate CA certificate.")
+      end
+    end
+
+    def sign(cert)
+      Rails.cache.delete(cache_key+'/certs')
```

Comment:
i understand, the main thing that bother me here is that you communicate directly with the "cache" vs thought the base class.
maybe instead we should keep a cache method on the base class